tests/reproducibility/test_graphsage.py

```diff
@@ -145,6 +145,67 @@ def gs_nai(
     return model
 
 
+def gs_link_pred_model(num_samples, generator, optimizer, bias, dropout, normalize):
+    layer_sizes = [50] * len(num_samples)
+    graphsage = GraphSAGE(
+        layer_sizes=layer_sizes,
+        generator=generator,
+        bias=bias,
+        dropout=dropout,
+        normalize=normalize,
+    )
+    # Build the model and expose input and output sockets of graphsage, for node pair inputs:
+    x_inp, x_out = graphsage.build()
+    pred = link_classification(
+        output_dim=1, output_act="relu", edge_embedding_method="ip"
+    )(x_out)
+
+    model = tf.keras.Model(inputs=x_inp, outputs=pred)
+
+    model.compile(
+        optimizer=optimizer, loss=tf.keras.losses.binary_crossentropy,
+    )
+
+    return model
+
+
+def gs_link_prediction(
+    g,
+    edge_ids,
+    edge_labels,
+    num_samples,
+    optimizer,
+    batch_size=4,
+    epochs=4,
+    bias=True,
+    dropout=0.0,
+    normalize="l2",
+    seed=0,
+    shuffle=True,
+):
+    set_seed(seed)
+    tf.random.set_seed(seed)
+    if shuffle:
```

Comment:
Why does `shuffle == True` require more than just `stellargraph.random.set_seed`? (I know this applies to other models in this file too, but I apparently missed asking about it until now.)

Reply:
It was introduced in #806 I mentioned it briefly in the description

> One slightly unfortunate thing is that using Model.fit_generator(..., shuffle=True) seems to rely on the global python random - so now we test that seeding the global python random is only needed when shuffle=True.

I'll double check if I get the same behaviour using `fit` instead.

Reply:
Ah; I'd forgotten about that.